Fixed documentation generation bug.

JordonPhillips opened the pull request:
This is a fix for #195. When attempting to detect lists, the path for the method was being examined instead of the path for the resource.

rayluo commented:
Thanks! We will review it.

kyleknap commented on boto3/docs/method.py:
```diff
@@ -46,7 +46,8 @@ def document_model_driven_resource_method(
             resource_action_model.resource.type)
 
-        if resource_action_model.path and '[]' in resource_action_model.path:
+        if (resource_action_model.resource.path and
```
So I think this logic that I originally put in place only accounts for a subset of the all of the possible occasions where a list of resources can be returned. After talking with @trevorrowe, the only true way to tell if a list or resources is returned is if the returned resource definition has at least one identifier that has a ``[]`` in its path. It would be cool if we can update the logic to that because your fix currently will not fix some other edge cases such as any of the create_tags actions. Here is the model https://github.com/boto/boto3/blob/develop/boto3/data/ec2/2014-10-01/resources-1.json#L528 and this is what the docs output: http://boto3.readthedocs.org/en/latest/reference/services/ec2.html#EC2.Image.create_tags which is incorrect.

kyleknap commented:
The PR looks good. I had one comment that will help fix some other issues with not displaying in the docs that a list of resource is returned.

JordonPhillips commented:
I updated the branch to address the extra edge cases. I abstracted out the collection detection logic into a hidden method, though I am unsure if that should be in `method.py` or `utils.py`. I left it in `method.py`, as that seemed best to me. I also updated the collections test case to properly verify collections are returned.

kyleknap commented on boto3/docs/method.py:
```diff
@@ -59,3 +59,14 @@ def document_model_driven_resource_method(
         new_return_section.style.new_line()
+
+
+def _method_returns_collection(resource):
```
I'm fine with keeping it in ``method.py`` because it is not needed in any other modules.

The name is a little misleading here. I would not call it as returning a collection. That is a whole other resource concept. This is more just checking if a list of resources is returned. Maybe call it ``_method_returns_resource_list()``?

kyleknap commented on boto3/docs/method.py:
```diff
@@ -59,3 +59,14 @@ def document_model_driven_resource_method(
+
+
+def _method_returns_collection(resource):
+    if resource.path and '[]' in resource.path:
```
I am pretty sure we can remove this check completely. Based on how resource models work, if a path with a ``[]`` is specified for that resource than at least one of the identifiers must have the ``[]`` in its path as well.

kyleknap commented:
Looks better. Had a couple of comments. Also I would not worry too much about the coveralls check not running. Looks like coveralls is down.

kyleknap commented:
Looks good. :ship:. I do not know what is up with coveralls, but tests pass so it should be good to merge.
